script.js

```diff
@@ -169,29 +193,64 @@ async function loadLocalVoices() {
   }
 }
 
+/**
+ * Adds a random interval, in milliseconds, into the
+ * `randomIntervalsInMilliseconds` number that may be used by the
+ * `speakAndWriteToDialogueText` function to create a more natural feeling,
+ * being applied whenever there is a white space in the input text.
+ */
+function addRandomIntervalInMilliseconds() {
+  const intervalInMilliseconds = calculateIntervalInMilliseconds();
+  const minimumIntervalValueInMilliseconds = 100;
+  const maximumIntervalValueInMilliseconds = 3e4 / intervalInMilliseconds;
+  const randomIntervalInMilliseconds =
+      Math.floor(Math.random() * maximumIntervalValueInMilliseconds) +
+      minimumIntervalValueInMilliseconds;
+
+  randomIntervalsInMilliseconds += randomIntervalInMilliseconds;
+}
+
+/**
+ * Cancels all the previous random intervals used, by reseting the
+ * `randomIntervalsInMilliseconds` to zero, as it was at the start.
+ */
+function cancelRandomIntervals() {
+  randomIntervalsInMilliseconds = 0;
+}
+
 /**
  * Creates and plays a speak with the input text inserted in the $inputText
- * element.
+ * element. It also, writes the text to the $dialogueText element at the same
+ * time it is speaking.
  *
  * @async
  */
-async function speak() {
+async function speakAndWriteToDialogueText() {
```

Comment:
```suggestion
async function speakAndWriteDialogue() {
```

This seems fine to me. Writing to the dialogue text = writing a dialogue. I hope this makes sense.